packages/flutter/test/painting/gradient_test.dart

```diff
@@ -263,6 +264,41 @@ void main() {
     ));
   });
 
+  test('RadialGradient lerp test with focal', () {
+    const RadialGradient testGradient1 = const RadialGradient(
+      center: Alignment.topLeft,
+      focal: Alignment.centerLeft,
+      radius: 20.0,
+      focalRadius: 10.0,
+      colors: const <Color>[
+        const Color(0x33333333),
+        const Color(0x66666666),
+      ],
+    );
+    const RadialGradient testGradient2 = const RadialGradient(
+      center: Alignment.topRight,
+      focal: Alignment.centerRight,
+      radius: 10.0,
+      focalRadius: 5.0,
+      colors: const <Color>[
+        const Color(0x44444444),
+        const Color(0x88888888),
+      ],
+    );
+
+    final RadialGradient actual = RadialGradient.lerp(testGradient1, testGradient2, 0.5);
+    expect(actual, const RadialGradient(
+      center: const Alignment(0.0, -1.0),
+      focal: const Alignment(0.0, 0.0),
+      radius: 15.0,
+      focalRadius: 7.5,
+      colors: const <Color>[
+        const Color(0x3B3B3B3B),
+        const Color(0x77777777),
+      ],
+    ));
+  });
```

Comment:
also add a test for lerp that verifies that when focal is null in both cases, the output has focal==null.

Reply:
So I think that's covered, but I added an explicit check for it.

One thing I'm less sure of now is `lerp` between a RadialGradient with a `focal` and one without.  I've added a test for that with the actual output - I think it looks right but does it make sense to you?

Reply:
tests look fine in principle

Reply:
Did a couple quick test and it looks like it should.  When focal is null it just basically is lerping back toward a regular RadialGradient.